server/libs/domain/src/partner/partner.core.ts

```diff
@@ -0,0 +1,33 @@
+import { BadRequestException, Logger } from '@nestjs/common';
+import { IPartnerRepository, PartnerDirection } from './partner.repository';
+import { PartnerEntity } from '@app/infra/entities';
+
+export class PartnerCore {
+  readonly logger = new Logger(PartnerCore.name);
+
+  constructor(private repository: IPartnerRepository) {}
+
+  getAll(userId: string, direction: PartnerDirection): Promise<PartnerEntity[]> {
+    return this.repository.getAll(userId, direction);
+  }
+
+  get(sharedBy: string, sharedWith: string): Promise<PartnerEntity | null> {
+    return this.repository.get(sharedBy, sharedWith);
+  }
+
+  create(sharedBy: string, sharedWith: string): Promise<PartnerEntity> {
+    return this.repository.create({
+      sharedBy,
+      sharedWith,
+    });
+  }
+
+  async remove(sharedBy: string, sharedWith: string): Promise<PartnerEntity> {
```

Comment:
Just a recommendation here - I have personally messed up a few times where I called a method like this with the arguments in the wrong order. It may be worth making an interface for this and taking advantage of the difference in key names, opposed to relying solely on the argument order. Especially since they (for me at least) are already somewhat easy to get mixed up.

```typescript
export interface PartnerId {
  sharedBy: string;
  sharedWith: string;
}

get({ sharedBy, sharedWith }: PartnerIds): ...
create({ sharedBy, sharedWith }: PartnerIds): ...
```